src/vcpkg/commands.set-installed.cpp

```diff
@@ -109,6 +244,14 @@ namespace vcpkg::Commands::SetInstalled
             }
         }
 
+        if (paths.manifest_mode_enabled() && paths.get_feature_flags().dependency_graph)
+        {
+            auto gh_root_dir = args.github_workspace.value_or_exit(VCPKG_LINE_INFO);
+            auto snapshot =
+                create_dependency_graph_snapshot(args, action_plan, paths.relative_path_to_manifest(gh_root_dir));
```

Comment:
This will crash if someone invokes vcpkg outside of the workspace root.

Reply:
In cases where something is wrong and would prevent us from doing the API submission (or if that fails), I agree it would be better to log and continue than to either fail entirely or crash. Ultimately, the dependency graph is a "value add", not the main reason someone would be running vcpkg.